src/Reflection/ReflectionParameter.php

```diff
@@ -504,14 +504,18 @@ public function getDefaultValueConstantName()
     public function getClass()
     {
         $hint = $this->getTypeHint();
-        if (!($hint instanceof Types\Object_)) {
+        if (!($hint instanceof Types\Object_  || $hint instanceof Types\Self_)) {
             return null;
         }
 
         if (!$this->reflector instanceof ClassReflector) {
             throw new \RuntimeException('Unable to reflect class type because we were not given a ClassReflector');
         }
 
+        if ($hint instanceof Types\Self_) {
```

Comment:
This can go before the `if (!$this->reflector instanceof ClassReflector) {` check